x-pack/legacy/plugins/siem/public/components/ml_popover/jobs_table/filters/jobs_table_filters.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
+
+import {
+  EuiFilterButton,
+  EuiFilterGroup,
+  EuiFlexGroup,
+  EuiFlexItem,
+  // @ts-ignore no-exported-member
+  EuiSearchBar,
+} from '@elastic/eui';
+import { EuiSearchBarQuery } from '../../../open_timeline/types';
+import * as i18n from './translations';
+import { JobsFilters, SiemJob } from '../../types';
+import { GroupsFilterPopover } from './groups_filter_popover';
+
+interface JobsTableFiltersProps {
+  siemJobs: SiemJob[];
+  onFilterChanged: Dispatch<SetStateAction<JobsFilters>>;
+}
+
+/**
+ * Collection of filters for filtering data within the JobsTable. Contains search bar, Elastic/Custom
+ * Jobs filter button toggle, and groups selection
+ *
+ * @param siemJobs jobs to fetch groups from to display for filtering
+ * @param onFilterChanged change listener to be notified on filter changes
+ */
+export const JobsTableFilters = React.memo<JobsTableFiltersProps>(
+  ({ siemJobs, onFilterChanged }) => {
+    const [filterQuery, setFilterQuery] = useState<string>('');
+    const [selectedGroups, setSelectedGroups] = useState<string[]>([]);
+    const [showCustomJobs, setShowCustomJobs] = useState<boolean>(false);
+    const [showElasticJobs, setShowElasticJobs] = useState<boolean>(false);
```

Comment:
Same things with these `useState`, you only need the template for the `string[]`

```ts
    const [filterQuery, setFilterQuery] = useState('');
    const [selectedGroups, setSelectedGroups] = useState<string[]>([]);
    const [showCustomJobs, setShowCustomJobs] = useState(false);
    const [showElasticJobs, setShowElasticJobs] = useState(false);
```

Optional though.

Reply:
I think I ended up adding the rest for consistency since `string[]` was required for `setSelectedGroups`. 

We should probably introduce a linter one way or another?